ddclient.in

```diff
@@ -5041,6 +5052,107 @@ sub nic_nfsn_update {
 
 ######################################################################
 
+######################################################################
+## nic_njalla_examples
+######################################################################
+sub nic_njalla_examples {
+    return <<"EoEXAMPLE";
+
+o 'njalla'
+
+The 'njalla' protocol is used by DNS service offered by njal.la.
+
+Configuration variables applicable to the 'njalla' protocol are:
+  protocol=njalla              ##
+  password=service-password    ## Generated password for your dynamic DNS record
+  quietreply=no|yes            ## If yes return empty response on success with status 200 but print errors
+  domain                       ## subdomain to update, use @ for base domain name, * for catch all
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=njalla                                             \\
+  password=njal.la-key
+  quietreply=no
+  domain.com
+
+EoEXAMPLE
+}
+######################################################################
+## nic_njalla_update
+##
+## written by satrapes
+##
+## based on https://njal.la/docs/ddns/
+## needs this url to update:
+## https://njal.la/update?h=host_name&k=domain_password&a=your_ip
+## response contains "code 200" on succesful completion
+######################################################################
+sub nic_njalla_update {
+    debug("\nnic_njalla_update -------------------");
+
+    foreach my $h (@_) {
+        # Read input params
+        my $ipv4 = delete $config{$h}{'wantipv4'};
+        my $ipv6 = delete $config{$h}{'wantipv6'};
+        my $quietreply = delete $config{$h}{'quietreply'};
+        my $ip_output = '';
+
+        # Build url
+        my $url = "https://$config{$h}{'server'}/update/";
+        $url .= "?h=";
+        $url .= $h;
+        $url .= "&k=";
+        $url .= $config{$h}{'password'};
```

Comment:
```suggestion
        my $url = "https://$config{$h}{'server'}/update/?h=$h&k=$config{$h}{'password'};
```

Reply:
You missed a `"` there but ok. Don't worry I will test it again with your changes. Apart from the IP v6 stuff because I don't have it enabled on my setup. But the logic is the same as IP v4 so not hugely concerned.